spec/site/alarm_spec.rb

```diff
@@ -67,4 +67,43 @@ def check_scripts
   #    system(SCRIPT_PATHS['deactivate_alarm'])
   #  end
   #end
+
+  it 'Disruption test', :script, sxl: '>=1.0.7' do |example|
+    check_scripts
+    component = COMPONENT_CONFIG['detector_logic'].keys.first
+
+    TestSite.isolated do |task,supervisor,site|
+      site.log "Checking connection", level: :test
+      # Is an explicit check needed?
+    end
+      
+    # Deactivate alarm 
+    system(SCRIPT_PATHS['activate_alarm'])
+
+    TestSite.isolated do |task,supervisor,site|
+      start_time = Time.now
+      site.log "Waiting for alarm", level: :test
+      message, response = nil,nil
+      expect do
+        response = site.wait_for_alarm task, component: component, aCId: 'A0302',
+          aSp: 'Issue', aS: 'Active', timeout: RSMP_CONFIG['alarm_timeout']
+      end.to_not raise_error, "Did not receive alarm"
+
+      delay = Time.now - start_time
+      site.log "alarm confirmed after #{delay.to_i}s", level: :test
```

Comment:
try using log_confirmation() from log_helpers.rb to handle the time measurement and logging. see other tests for how it's used.